bookstore/tests/test_archive.py

```diff
@@ -19,7 +21,34 @@ def test_create_contentsarchiver_invalid_args_count():
     archiver = BookstoreContentsArchiver()
     assert archiver
 
-    record = ArchiveRecord('workspace/project', 'my_notebook', 100.2)
+    record = ArchiveRecord('my_notebook_path.ipynb', json.dumps(new_notebook()), 100.2)
     assert record
 
     await archiver.archive(record)
+
+
+@pytest.mark.asyncio
+async def test_archive_abort_with_lock():
+    archiver = BookstoreContentsArchiver()
+    record = ArchiveRecord('my_notebook_path.ipynb', json.dumps(new_notebook()), 100.2)
```

Comment:
Some extra comments I'd love to see added to this test:

```
Acquire a lock in advance so that when the archiver attempts to archive, it will abort.
```

Not sure how we can test to make sure that it really did abort here -- mocking the log? Returning a boolean? Verifying that `put_object` wasn't called by mocking s3?

Reply:
I just learned about pytests logging capture fixture (https://docs.pytest.org/en/latest/logging.html#caplog-fixture)… it makes the logs really easy to capture.